app/controllers/financialOverview/psa/AllPenaltiesAndChargesController.scala

```diff
@@ -101,7 +99,6 @@ class AllPenaltiesAndChargesController @Inject()(
 
       psaPenaltiesAndChargesService.getPenaltiesForJourney(request.idOrException, journeyType).flatMap { penaltiesCache =>
 
-        val allChargeRefs = penaltiesCache.penalties.map(_.chargeReference)
         val title: Message = Message("penalties.nonAft.title", Message(s"penaltyType.${penaltyType.toString}"), year)
         val filteredPenalties: Seq[PsaFSDetail] = penaltiesCache.penalties
           .filter(_.periodEndDate.getYear == year.toInt)
```

Comment:
if you filter the _.periodEndDate  then you could remove the if (filteredPenalties.nonEmpty)
see below
.filter(_.periodEndDate.nonEmpty)